docs/guides/connecting_to_your_data/filesystem/spark.md

```diff
@@ -2,4 +2,152 @@
+
+<Tabs
+  groupId="yaml-or-python"
+  defaultValue='yaml'
+  values={[
+  {label: 'YAML', value:'yaml'},
+  {label: 'python', value:'python'},
+  ]}>
+  <TabItem value="yaml">
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/filesystem/spark_yaml_example.py#L20-L38
+```
+
+Run this code to test your configuration.
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/filesystem/spark_yaml_example.py#L44
+```
+
+</TabItem>
+<TabItem value="python">
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/filesystem/spark_python_example.py#L21-L39
+```
+
+Run this code to test your configuration.
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/filesystem/spark_python_example.py#L45
+```
+
+</TabItem>
+</Tabs>
+
+If you specified a path containing CSV files you will see them listed as `Available data_asset_names` in the output of `test_yaml_config()`.
+
+Feel free to adjust your configuration and re-run `test_yaml_config()` as needed.
+
+### 4. Save the Datasource configuration to your DataContext
+
+Save the configuration into your `DataContext` by using the `add_datasource()` function.
+
+<Tabs
+  groupId="yaml-or-python"
+  defaultValue='yaml'
+  values={[
+  {label: 'YAML', value:'yaml'},
+  {label: 'python', value:'python'},
+  ]}>
+  <TabItem value="yaml">
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/filesystem/spark_yaml_example.py#L46
+```
+
+</TabItem>
+<TabItem value="python">
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/filesystem/spark_python_example.py#L49
+```
+
+</TabItem>
+</Tabs>
+
+### 5. Test your new Datasource
+
+Verify your new Datasource by loading data from it into a `Validator` using a `BatchRequest`.
+
+<Tabs
+  defaultValue='runtime_batch_request'
+  values={[
+  {label: 'Specify a path to single CSV', value:'runtime_batch_request'},
+  {label: 'Specify a data_asset_name', value:'batch_request'},
+  ]}>
+  <TabItem value="runtime_batch_request">
+
+Add the path to your CSV in the `path` key under `runtime_parameters` in your `BatchRequest`.
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/filesystem/spark_yaml_example.py#L49-L55
+```
+Then load data into the `Validator`.
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/filesystem/spark_yaml_example.py#L61-L66
+```
+
+  </TabItem>
+  <TabItem value="batch_request">
+
+Add the name of the data asset to the `data_asset_name` in your `BatchRequest`.
+
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/filesystem/spark_yaml_example.py#L81-L85
+```
+Then load data into the `Validator`.
+```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/filesystem/spark_yaml_example.py#L93-L99
+```
+
```

Comment:
I think these lines may need to be double-checked

Reply:
Good catches!